apps/E2E/src/common/BasePage.ts

```diff
@@ -159,15 +159,20 @@ export class BasePage {
 
   /* Waits for the primary UI test element to be displayed. If the element doesn't load before the timeout, it causes the test to fail. */
   async waitForPrimaryElementDisplayed(timeout?: number): Promise<void> {
-    await browser.waitUntil(async () => await (await this._primaryComponent).isDisplayed(), {
+    await this.waitForElementDisplayed(await this._primaryComponent, timeout);
+  }
+
+  /* Waits for the the specified UI test element to be displayed. If the element doesn't load before the timeout, it causes the test to fail. */
+  async waitForElementDisplayed(component: WebdriverIO.Element, timeout?: number): Promise<void> {
+    await browser.waitUntil(async () => await (await component).isDisplayed(), {
```

Comment:
I don't believe we need 2 awaits here, it should be `await component.isDisplayed()`. Because component is a `WebDriverIO.Element`, not a Promise. `await` is only needed with a Promise